packages/backend/src/tenants/model.ts

```diff
@@ -8,4 +9,22 @@ export class Tenant extends BaseModel {
   public email!: string
   public apiSecret!: string
   public publicName?: string
+
+  public deletedAt?: Date
+
+  $formatJson(json: Pojo): Pojo {
+    json = super.$formatJson(json)
+    return {
+      ...json,
+      deletedAt: json.deletedAt.toISOString()
+    }
+  }
+}
+
+export type TenantWithIdpConfig = Pick<
+  Tenant,
+  'id' | 'email' | 'apiSecret' | 'publicName' | 'createdAt' | 'updatedAt'
+> & {
+  idpConsentUrl: string
+  idpSecret: string
```

Comment:
Do we need `idpConsentUrl` and `idpSecret` in backend?

Reply:
> We need this in backend, so that with one call, we add tenant data on both backend, and auth service.

I'm guessing you are referring to this comment @golobitch?

Also, I see, we are not storing the `idpConsentUrl` and `idpSecret` in the tenants table, (`TenantWithIdpConfig` is just a type). At first, _not_ storing the data makes sense to me (we shouldn't care about IDP stuff in `backend`). However, how would that work for doing pagination over tenants? If we don't store these, then we would need to call `getTenant` in the auth service for every tenant result?

Reply:
Ah yes, @njlie, @mkurapov has a point here :)

Reply:
Perhaps we should treat the backend's `idpConsentUrl` and `idpSecret` as some sort of cache-like thing. As in for pagination, we return the backend's entries for each tenant row, but in the query for an individual tenant, make the call to the authorization server. Whenever a tenant is mutated such that the IDP fields are changed, it would update the columns in both backend and auth.

Reply:
> but in the query for an individual tenant, make the call to the authorization server

The fields should basically always be the same, so I think calling to the AS isn't needed. The updates shouldn't happen frequently, and if they do, they'll be stored in the backend tenants table anyway as you describe